parserator_web/static/js/index.js

```diff
@@ -1,2 +1,50 @@
-/* TODO: Flesh this out to connect the form to the API and render results
-   in the #address-results div. */
+/* TODO: Flesh this out to connect the form to the API and render results in the #address-results div. */
+
+//Stores the user's input and passes it to the server that then renders data in which the function can utilize to populate page with said data or message of an error.
+function getUserInput(event) {
+   //Prevents refreshing of the page
+   event.preventDefault()
+   //Stores DOM elements in variables to be later modified
+   const table = document.querySelector("#address-results > table > tbody");
+   const addressTypeHTML = document.querySelector("#address-results > p");
+   //Clears previously rendered data
+   table.innerHTML = ""
+   addressTypeHTML.innerHTML = "Address type: "
+   //Stores the user's input.
+   let address = document.getElementById("address").value;
+
+   //Runs get request. Sends address input to server to render a response.
+   fetch(`http://localhost:8000/api/parse/?csrfmiddlewaretoken=8cvhuDLNK2wPx7zFR9OnGgXD0hw8gTCnCrml2hkYxDLLwWRnqiC0fLKbDLPZxKER&address=${address}`)
```

Comment:
this works locally, but won't work on staging or production. one way to address this is to make the call a relative call like this

```javascript
fetch(`api/parse/?address=${address}`)
```
note: you don't need the `csrfmiddlewaretoken` parameter.

you might also like to uses [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) to handle building your parameter arguments.

```javascript
parameters = new URLSearchParams({address: address})
fetch(api/parse/? + parameters.toString())
```

Reply:
Ah, that's right! I should have used a relative call. 

Appreciate the correction on the `csrfmiddlewaretoken`  parameter!

Appreciate the nudge to the URLSearchParams! Just read into it and definitely will be adding it to my toolkit!